lib/pulse_core/PulseCore.MonotonicStateMonad.fsti

```diff
@@ -0,0 +1,76 @@
+module PulseCore.MonotonicStateMonad

```

Comment:
This provides a value-type interface for an abstract indexed monad support monotonic state. This makes things a bit easier and more direct in defining the PulseCore.Semantics, rather than going via an indexed effect